apps/mobile/src/features/token/utils/render-collectible-details.tsx

```diff
@@ -0,0 +1,120 @@
+import { Linking } from 'react-native';
+
+import { SummaryTableItem, SummaryTableRoot } from '@/components/summary-table';
+import { t } from '@lingui/core/macro';
+
+import { Pressable, Text } from '@leather.io/ui/native';
+import { truncateMiddle } from '@leather.io/utils';
+
+import { TokenDetailsCard } from '../components/token-details-card';
+
+function formatLabel(key: string): string {
+  // Special case for locationUrl in collection context
+  if (key === 'location_url') {
+    return t`Collection URL`;
+  }
+
+  // Replace underscores with spaces
+  let formatted = key.replace(/_/g, ' ');
+
+  // Add spaces between PascalCase words
+  formatted = formatted.replace(/([a-z])([A-Z])/g, '$1 $2');
+
+  // Capitalize first letter
+  return formatted.charAt(0).toUpperCase() + formatted.slice(1);
+}
+
+function renderValue(value: any, key: string): React.ReactNode {
```

Comment:
```suggestion
function renderValue(value: unknown, key: string): React.ReactNode {
```

better to use unknown if narrowing